url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/Trading_Engine/myDatabase.java

```diff
@@ -0,0 +1,571 @@
+			BufferedReader br = new BufferedReader(new FileReader(f));
+			String head = "";
+			if((head=br.readLine())!=null){
+				//System.out.println("first character of head: " + head.charAt(0));
+				if(head.charAt(0)== '#'){
+					String[] tableElement = head.split(",");
+					tableElement[3] = "Record_Type";
+					tableElement[6] = "Undisclosed_Volume";
+					tableElement[9] = "Trans_ID";
+					tableElement[10] = "Bid_ID";
+					tableElement[11] = "Ask_ID";
+					tableElement[12] = "Bid_Ask";
+					tableElement[13] = "Entry_Time";
+					tableElement[14] = "Old_Price";
+					tableElement[15] = "Old_Volume";
+					tableElement[16] = "Buyer_Broker_ID";
+					tableElement[17] = "Seller_Broker_ID";
+					String tableQuery = tableElement[0].substring(1) + " varchar(3), " +
+							tableElement[1] + " date, " +
+							tableElement[2] + " time, " +
+							"millisecond" + " integer, "+
+							tableElement[3] + " text, " +
+							tableElement[4] + " double, " +
+							tableElement[5] + " integer, " +
+							tableElement[6] + " integer, " +
+							tableElement[7] + " float, " +
+							tableElement[8] + " text, " +
+							tableElement[9] + " integer, " +
+							tableElement[10] + " bigint, " +
+							tableElement[11] + " bigint, " +
+							tableElement[12] + " varchar(1), " +
+							tableElement[13] + " time, " +
+							tableElement[14] + " double, " +
+							tableElement[15] + " integer, " +
+							tableElement[16] + " integer, " +
+							tableElement[17] + " integer ";
+					String newHead = tableElement[0].substring(1);
+					for(int i = 1; i < tableElement.length; i++){
+						newHead += " , " + tableElement[i];
+						if(i == 2){
+							newHead += " , " + "millisecond";
+						}
+					}
+					//System.out.println(tableQuery);
+					s.execute("create table " + dbName + "(" + tableQuery + ")");
+					s.execute("create table old_trade_list(" + tableQuery + ")");
+					insertAllDatabase(connection,br,newHead,dbName,overviewResult);
+				} else {
+					System.out.println("Error: file content not csv format");
+				}
+			}else {
+				System.out.println("Error: file is empty");
+			}
+			br.close();
+			s.close();
+		}catch (Exception e) {
+			System.out.println("In insertAll:  " + e);
+		}
+		return overviewResult;
+	}
+	public void deleteAllTables(){
+		try {
+			Statement statement = connection.createStatement();
+			DatabaseMetaData dbm = connection.getMetaData();
+			String[] types = {"TABLE"};
+			ResultSet allTables = dbm.getTables(null,null,"%",types);
+			if(allTables!=null){
+				while(allTables.next()){
+					String table = allTables.getString("TABLE_NAME");
+					statement.addBatch("drop table " + table);
+					System.out.println("delete table: " + table);
+				};
+				statement.executeBatch();
+			}
+			allTables.close();
+			statement.close();
+		}catch(Exception e) {
+			System.out.println("in deleteAllTables: " + e);
+		}
+	}
+	public static void insertQuery(PreparedStatement pstmt, String[] insertElement){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 27. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.